components/suggest/src/store.rs

```diff
@@ -4426,13 +4531,67 @@ mod tests {
 
         let store = unique_test_store(SnapshotSettingsClient::with_snapshot(snapshot));
         store.dbs()?.writer.write(|dao| {
-            dao.put_meta(LAST_INGEST_META_KEY, 30)?;
+            dao.put_meta("last_quicksuggest_ingest_data", 30)?;
             Ok(())
         })?;
         store.ingest(SuggestIngestionConstraints::default())?;
 
         store.dbs()?.reader.read(|dao| {
-            assert_eq!(dao.get_meta::<u64>(LAST_INGEST_META_KEY)?, Some(30));
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_data")?,
+                Some(30)
+            );
+            Ok(())
+        })?;
+
+        Ok(())
+    }
+
+    /// Tests that we only ingest providers that we're concerned with.
+    #[test]
+    fn ingest_constraints_provider() -> anyhow::Result<()> {
+        before_each();
+
+        let snapshot = Snapshot::with_records(json!([{
+            "id": "data-1",
+            "type": "data",
+            "last_modified": 15,
+        }, {
+            "id": "icon-1",
+            "type": "icon",
+            "last_modified": 30,
+        }]))?;
+
+        let store = unique_test_store(SnapshotSettingsClient::with_snapshot(snapshot));
+        store.dbs()?.writer.write(|dao| {
+            // Check that existing data is updated properly.
+            dao.put_meta("last_quicksuggest_ingest_data", 10)?;
+            Ok(())
+        })?;
+
+        let constraints = SuggestIngestionConstraints {
+            max_suggestions: Some(100),
+            providers: Some(vec![SuggestionProvider::Amp, SuggestionProvider::Pocket]),
+        };
+        store.ingest(constraints)?;
+
+        store.dbs()?.reader.read(|dao| {
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_data")?,
+                Some(15)
+            );
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_icon")?,
+                Some(30)
+            );
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_pocket-suggestions")?,
+                None
+            );
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_amo-suggestions")?,
```

Comment:
Let's also test other suggestion types (`Yelp, AmpMobile`, `Mdn`, and `Weather`) were not ingested for completeness.